File: lab9.cpp
#include <iostream>
#include <stdio.h>
#include <cstring>
using namespace std;
int* counter(char[]);

int main()
{
    cout << "enter the string up to 100 symbols:\n";
    char str[100];
    gets(str);
    int* amount = counter(str);
    for(int i = 0; i < 255; i++)
    {
        if(amount[i] != 0) cout << "char [" << char(i) << "] is found " << amount[i] << " times\n"; 
    }
}

int* counter(char str[])
{
    int length = strlen(str);
    int *char_count = new int[255];
    for(int i = 0; i < length; i++)
    {
        char_count[(int)str[i]]++;
    }
    delete [] char_count;
    return char_count;
}

File: lab9.py
def input_to_list():
    symbols = []
    empty_line = True
    while empty_line:
        input_string = input("Enter your string \n")
        if input_string is not None:
            empty_line = False
    for i in input_string:
        symbols.append(i)
    return symbols


def display_the_number_of_symbols(symbols):
    new_symbol_list = []
    for each_symbol in symbols:
        if each_symbol not in new_symbol_list:
            new_symbol_list.append(each_symbol)
    for each_final_symbol in new_symbol_list:
        print("Symbol ", each_final_symbol, "found ", symbols.count(each_final_symbol), " times")